pkg/controller/middleware/auth.go

```diff
@@ -166,89 +169,31 @@ func (rah *RequireAuthHandler) Handle(next http.Handler) http.Handler {
-// a user exists in the database, and that said user has not made too many
-// requests
-func QuotaKeyIssuance(ctx context.Context, client *auth.Client, db *database.Database, ttl time.Duration, requestsPerTTL int) *QuotaKeyIssuanceHandler {
-	logger := logging.FromContext(ctx)
-	return &QuotaKeyIssuanceHandler{ctx, client, db, ttl, requestsPerTTL, logger}
-}
-
-func (rah *QuotaKeyIssuanceHandler) Handle(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if err := func() error {
-			// Get the cookie
-			cookie, err := r.Cookie("session")
-			if err != nil {
-				return fmt.Errorf("failed to get cookie: %w", err)
-			}
-
-			// Verify cookie
-			token, err := rah.client.VerifySessionCookie(rah.ctx, cookie.Value)
-			if err != nil {
-				return fmt.Errorf("failed to verify session cookie: %w", err)
-			}
-
-			// Get the email
-			emailRaw, ok := token.Claims["email"]
-			if !ok {
-				// s.DestroySession(c) // TODO
-				return fmt.Errorf("session is missing email")
-			}
-
-			// Convert to string
-			email, ok := emailRaw.(string)
-			if !ok {
-				return fmt.Errorf("email is not a string")
-			}
-
-			// Lookup the user by email
-			user, err := rah.db.FindUser(email)
-			if err != nil || user == nil {
-				return ErrUserNotFound
-			}
-
-			// Check if a quota window has lapsed and reset if necessary.
-			if time.Now().After(user.LastQuotaReset.Add(rah.quotaTTL)) {
-				user.LastQuotaReset = time.Now()
-				user.IssuancesAgainstQuota = 0
-			}
-
-			user.IssuancesAgainstQuota++
-			user.VerificationCodesIssued++
-
-			// Save before checking quota so we have a record of attempted
-			// creations by a user account.
-			if err := rah.db.SaveUser(user); err != nil {
-				return fmt.Errorf("failed to update user for quota: %w", err)
-			}
-
-			// Check if a user has exhausted their quota
-			if user.IssuancesAgainstQuota > rah.requestsPerTTL {
-				return fmt.Errorf("no remaining quota for current user")
-			}
-
-			return nil
-		}(); err != nil {
-			rah.logger.Errorw("QuotaKeyIssuance", "error", err)
-
-			if controller.IsJSONContentType(r) {
-				controller.WriteJSON(w, http.StatusTooManyRequests, nil)
-			} else {
-				flash.FromContext(w, r).Error("Quota Exceeded or Unauthorized")
-				http.Redirect(w, r, "/signout", http.StatusFound)
-			}
-		} else {
-			next.ServeHTTP(w, r)
+func userIdKeyFunc() httplimit.KeyFunc {
+	return func(r *http.Request) (string, error) {
+		rawUser, ok := httpcontext.GetOk(r, "user")
+		if !ok {
+			// flash.FromContext(w, r).Error("Unauthorized")
+			// http.Redirect(w, r, "/signout", http.StatusFound)
+			return "", fmt.Errorf("unauthorized")
 		}
-	})
+		user, ok := rawUser.(*database.User)
+		if !ok {
+			// flash.FromContext(w, r).Error("internal error - you have been logged out.")
+			// http.Redirect(w, r, "/signout", http.StatusFound)
+			return "", fmt.Errorf("internal error")
+		}
+		return user.Email, nil
```

Comment:
I recommend we hash this (sha256 is probably fine). Otherwise we risk putting PII in redis.